compiler/ast.nim

```diff
@@ -1626,17 +1626,24 @@ proc shallowCopy*(src: PNode): PNode =
 
 proc copyTree*(src: PNode): PNode =
   # copy a whole syntax tree; performs deep copying
-  copyNodeImpl(result, src):
-    newSeq(result.sons, src.len)
-    for i in 0..<src.len:
-      result[i] = copyTree(src[i])
+  var stack = @[(addr result, src)]
+  while stack.len > 0:
+    let (target, source) = stack.pop()
+    copyNodeImpl(target[], source):
+      newSeq(target.sons, source.len)
+      for i in 0..<source.len:
+        stack.add((addr target.sons[i], source[i]))
```

Comment:
We will probably need a way to add capacity into the seq or this `.add` will slow everything down :(